perfkitbenchmarker/providers/azure/azure_disk.py

```diff
@@ -104,80 +109,73 @@ def __init__(self, disk_spec, vm_name, machine_type):
 
   def _Create(self):
     """Creates the disk."""
+    assert not self.is_image
 
     if self.disk_type == PREMIUM_STORAGE:
       assert FLAGS.azure_storage_type == azure_flags.PLRS
     else:
       assert FLAGS.azure_storage_type != azure_flags.PLRS
 
     with self._lock:
-      create_cmd = [AZURE_PATH,
-                    'vm',
-                    'disk',
-                    'attach-new',
-                    '--host-caching=%s' % self.host_caching,
-                    self.vm_name,
-                    str(self.disk_size)]
-      vm_util.IssueRetryableCommand(create_cmd)
-
-      if self.vm_name not in AzureDisk.num_disks:
-        AzureDisk.num_disks[self.vm_name] = 0
-      self.lun = AzureDisk.num_disks[self.vm_name]
-      AzureDisk.num_disks[self.vm_name] += 1
+      _, _, retcode = vm_util.IssueCommand(
+          [azure.AZURE_PATH, 'vm', 'disk', 'attach-new',
+           '--host-caching', self.host_caching,
+           self.vm_name, str(self.disk_size)] +
+          self.resource_group.args)
+
+      if retcode:
+        raise errors.Resource.RetryableCreationError(
+            'Error creating Azure disk.')
+
       self.created = True
```

Comment:
Why do some Resource._Create implementations do `self.created = True` while others don't (like `AzureResourceGroup`)? Is it because `AzureResourceGroup` doesn't support creation retries?